File: audit/tasks/aril-novel/shape_area/controls/shape_area.ts
// Reference-language control (TypeScript). Statement only — not machine-validated
// in this environment (no Node/tsc). Same I/O contract as the task.
type Shape = { kind: "circle"; r: number } | { kind: "rect"; w: number; h: number };

function area(s: Shape): number {
  switch (s.kind) {
    case "circle": return 3 * s.r * s.r;
    case "rect": return s.w * s.h;
  }
}

const shapes: Shape[] = [{ kind: "circle", r: 2 }, { kind: "rect", w: 3, h: 5 }];
for (const s of shapes) console.log(area(s));

File: audit/tasks/aril-novel/shape_area/controls/shape_area.rs
enum Shape {
    Circle(i32),
    Rect(i32, i32),
}

fn area(s: &Shape) -> i32 {
    match s {
        Shape::Circle(r) => 3 * r * r,
        Shape::Rect(w, h) => w * h,
    }
}

fn main() {
    let shapes = [Shape::Circle(2), Shape::Rect(3, 5)];
    for s in &shapes {
        println!("{}", area(s));
    }
}
